spanner/stats.go

```diff
@@ -167,3 +210,108 @@ func EnableStatViews() error {
 		ReleasedSessionsCountView,
 	)
 }
+
+// EnableGfeLatencyView enables GFELatency metric
+func EnableGfeLatencyView() error {
+	GFELatencyMetricsEnabled = true
+	return view.Register(GFELatencyView)
+}
+
+// EnableGfeHeaderMissingCountView enables GFEHeaderMissingCount metric
+func EnableGfeHeaderMissingCountView() error {
+	GFELatencyMetricsEnabled = true
+	return view.Register(GFEHeaderMissingCountView)
+}
+
+// EnableGfeLatencyAndHeaderMissingCountViews enables GFEHeaderMissingCount and GFELatency metric
+func EnableGfeLatencyAndHeaderMissingCountViews() error {
+	GFELatencyMetricsEnabled = true
+	return view.Register(
+		GFELatencyView,
+		GFEHeaderMissingCountView,
+	)
+}
+
+// DisableGfeLatencyAndHeaderMissingCountViews disables GFEHeaderMissingCount and GFELatency metric
+func DisableGfeLatencyAndHeaderMissingCountViews() {
+	GFELatencyMetricsEnabled = false
+	view.Unregister(
+		GFELatencyView,
+		GFEHeaderMissingCountView,
+	)
+}
+
+func captureGFELatencyStats(ctx context.Context, md metadata.MD, keyMethod string) error {
+	if len(md.Get("server-timing")) == 0 {
+		recordStat(ctx, GFEHeaderMissingCount, 1)
+		return nil
+	}
+	serverTiming := md.Get("server-timing")[0]
+	gfeLatency, err := strconv.Atoi(strings.TrimPrefix(serverTiming, "gfet4t7; dur="))
+	if !strings.HasPrefix(serverTiming, "gfet4t7; dur=") || err != nil {
+		return err
+	}
+	// Record GFE latency with OpenCensus.
+	ctx = tag.NewContext(ctx, tag.FromContext(ctx))
+	ctx, err = tag.New(ctx, tag.Insert(tagKeyMethod, keyMethod))
+	if err != nil {
+		return err
+	}
+	recordStat(ctx, GFELatency, int64(gfeLatency))
+	return nil
+}
+
+func checkCommonTagsGFELatency(t *testing.T, m map[tag.Key]string) {
```

Comment:
nit: consider moving this method to one of the `_test.go` files, as it seems to be a test method.